File: android-mobile-development/期末课设/Company/app/src/main/java/com/example/company/News.java
package com.example.company;

import androidx.annotation.NonNull;

public class News {
    private final String name;
    private final String content;
    private final int imageResourceId;

    public News(String name, String content, int imageResourceId) {
        this.name = name;
        this.content = content;
        this.imageResourceId = imageResourceId;
    }

    public static final News[] news = {
            new News("news one","aaaaaaaaaaaaaaaaaa",R.drawable.news01),
            new News("news two","bbbbbbbbbbbbbbbbbb",R.drawable.news02)
    };
    @NonNull
    public String toString()
    {
        return name;
    }

    public String getName() {
        return name;
    }

    public String getContent() {
        return content;
    }

    public int getImageResourceId() {
        return imageResourceId;
    }
}

File: android-mobile-development/final_course_project/Company/app/src/main/java/com/example/company/data/entity/News.kt
package com.example.company.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "NEWS")
data class News(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id") val id: Int = 0,
    @ColumnInfo(name = "TITLE") val title: String,
    @ColumnInfo(name = "IMAGE_RESOURCE_ID") val imageResourceId: Int,
    @ColumnInfo(name = "CONTENT") val content: String
)
